backend/Auth/ResetPassword.ts

```diff
@@ -115,6 +119,63 @@ router.get<Params, ResponseBody, QueryParams>(
   }
 );
 
+router.put(
+  "/api/passwords",
+  body("password").isStrongPassword(),
+  body("token").exists(),
+  async (req, res) => {
+    const { password, token } = req.body;
+
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      if (token) {
+        return invalidRequest(
+          res,
+          "Password must be at least 8 characters long, contain one number, one uppercase letter, and one special character."
+        );
+      } else {
+        return invalidRequest(
+          res,
+          "Request must include token. Please reset your password again."
+        );
+      }
+    }
+
+    if (!tokenIsValid(token, jwtSecret)) {
+      return invalidRequest(res, "Invalid token");
+    }
+
+    const rows = await JWTModel.findAll({
+      where: {
+        token: token,
+      },
+    });
+
+    if (rows.length >= 1) {
+      const userId = rows[0].userId;
+      const hash = await bcrypt.hash(password, 5);
+
+      const toUpdate = {
+        password: hash,
+      };
+
+      const [numUpdated] = await UserModel.update(toUpdate, {
+        where: {
+          id: userId,
+        },
+      });
+
+      if (numUpdated === 0) {
+        notFound(res, `Unable to update password`);
```

Comment:
I think this should be an http 500 Server Error rather than a 404 Not Found. The token's user id came from our database, so if that user id doesn't exist in the users table then that's a bug in our system. 500 communicates "server's fault" whereas 404 communicates "requester's fault." And in that case, it would be the server's fault for having a database in a bad state.